_posts/2018-11-08-google-cpp-style-guide-is-no-good.md

```diff
@@ -0,0 +1,297 @@
+---
+layout: post
+title: Google C++ Style Guide is No Good
+date: '2018-11-06T20:59:00.000-07:00'
+author: Eugene Yakubovich
+tags:
+- C++
+modified_time: '2018-10-23T20:59:46.484-07:00'
+---
+
+The [Google Style Guide (GSG)](https://google.github.io/styleguide/cppguide.html) for C++ has become popular outside of Google.
+But I don't like it at all. There... I said it.
+I want to take this opportunity to explain why I feel so strongly about it.
+
+## GSG "style"
+The general observation is that the style guide is written in a prohibitive fashion.
+Unlike C++ Core Guidelines that try to explain how to use the language effectively, GSG is about forbidding the use of certain features.
+Many of the rules prohibit the use of a feature over the fear of confusion, abuse, ambiguity and bugs.
+It is a well known fact that every feature (in every language) can be abused and misunderstood.
+Every line of code (and feature) can be a source of bugs.
+At the same time, every feature that went into the language was vetted for being useful.
+It was deemed that without the said feature, the code was significantly worse (e.g more verbose, prone to bugs).
```

Comment:
originally,  but are older features pruned ? Normally from what I remember C++ tends to keep every old feature as to not break old code.

Reply:
Yeah, C++ deprecates/removes very few features. At the same time, I can think of *very* few features that are now *completely* obsolete. It's just that they are less useful now because there are alternatives for some of the cases in which they were used before.

But I'll try to put a sentence to this point.